habitat_sim/utils/sim_utils.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import habitat_sim
+
+VisualSensorTypeSet = {
+    habitat_sim.SensorType.COLOR,
+    habitat_sim.SensorType.DEPTH,
+    habitat_sim.SensorType.SEMANTIC,
+}
+CameraSensorSubTypeSet = {
+    habitat_sim.SensorSubType.PINHOLE,
+    habitat_sim.SensorSubType.ORTHOGRAPHIC,
+}
+
+
+def remove_all_objects(sim):
+    r"""Removes all objects in simulator instance
+    :param sim: Instance of Simulator
+    """
+    for obj_id in sim.get_existing_object_ids():
+        sim.remove_object(obj_id)
+
+
+def make_sensor_specs_from_settings(sensors, settings):
+    r"""Creates and returns a list of initialized SensorSpecs
+    :param sensors: Dictionary of sensors to initialize and relevant parameters, sensor_type and sensor_subtype are mandatory for each sensor
+    :param settings: Dictionary modifying parameters and toggling sensors as true/false
+    :return: List[hsim.SensorSpec] of initialized SensorSpecs
+    """
+    sensor_specs = []
+    for sensor_uuid, sensor_params in sensors.items():
+        if settings[sensor_uuid]:
+            # Check if type VisualSensorSpec
+            if sensor_params["sensor_type"] not in VisualSensorTypeSet:
+                # TODO: Add more checks for NonVisualSensorSpec
+                raise ValueError(
+                    f"""{sensor_params["sensor_type"]} is an illegal sensorType that is not implemented yet"""
+                )
+            else:
+                # Check if type CameraSensorSpec
+                if sensor_params["sensor_subtype"] not in CameraSensorSubTypeSet:
+                    # TODO: Add more checks for other types of sensors
+                    raise ValueError(
+                        f"""{sensor_params["sensor_subtype"]} is an illegal sensorSubType for a VisualSensor"""
+                    )
+                else:
+                    sensor_spec = habitat_sim.CameraSensorSpec()
```

Comment:
If the subtype is "pinhole" or "ortho", it will use CameraSensorSpec() to initialize it.

Reply:
Yes! That is what we're doing here :)
clang-tidy formatting has it so that the errors should go before the valid cases, that's why we raise the ValueError if subtype is not pinhole or ortho, and use CameraSensorSpec() to initialize otherwise

Reply:
No reason to havee the else then because raising the Value will terminate the function. No reason to waste white space.

Reply:
@vauduong : no, you do not understand what I am talking about.
In the future there would be `fisheyeSensorSpec`, not just `CameraSensorSpec`. You have to have the if clause to control which one to use.

Reply:
I see, thank you, I will change that then!

Reply:
I fixed it based on my understanding in the latest commit. Take a look and let me know if I got everything right.